zio-schema/shared/src/main/scala/zio/schema/DynamicValue.scala

```diff
@@ -1866,4 +1871,419 @@ object DynamicValue {
+      ),
+      _.asInstanceOf[DynamicValue.DynamicAst]
+    )
+
+  private val singletonCase: Schema.Case[DynamicValue.Singleton[Any], DynamicValue] =
+    Schema.Case(
+      "Singleton",
+      Schema[Unit].transform(_ => DynamicValue.Singleton(()), _ => ()),
+      _.asInstanceOf[DynamicValue.Singleton[Any]]
+    )
+
+  private val primitiveUnitCase: Schema.Case[DynamicValue.Primitive[Unit], DynamicValue] =
+    Schema.Case(
+      "Unit",
+      Schema.primitive[Unit].transform(unit => DynamicValue.Primitive(unit, StandardType[Unit]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Unit]]
+    )
+
+  private val primitiveStringCase: Schema.Case[DynamicValue.Primitive[String], DynamicValue] =
+    Schema.Case(
+      "String",
+      Schema.primitive[String].transform(s => DynamicValue.Primitive(s, StandardType[String]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[String]]
+    )
+
+  private val primitiveBooleanCase: Schema.Case[DynamicValue.Primitive[Boolean], DynamicValue] =
+    Schema.Case(
+      "Boolean",
+      Schema.primitive[Boolean].transform(b => DynamicValue.Primitive(b, StandardType[Boolean]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Boolean]]
+    )
+
+  private val primitiveShortCase: Schema.Case[DynamicValue.Primitive[Short], DynamicValue] =
+    Schema.Case(
+      "Short",
+      Schema.primitive[Short].transform(sh => DynamicValue.Primitive(sh, StandardType[Short]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Short]]
+    )
+
+  private val primitiveIntCase: Schema.Case[DynamicValue.Primitive[Int], DynamicValue] =
+    Schema.Case(
+      "Int",
+      Schema.primitive[Int].transform(i => DynamicValue.Primitive(i, StandardType[Int]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Int]]
+    )
+
+  private val primitiveLongCase: Schema.Case[DynamicValue.Primitive[Long], DynamicValue] =
+    Schema.Case(
+      "Long",
+      Schema.primitive[Long].transform(l => DynamicValue.Primitive(l, StandardType[Long]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Long]]
+    )
+
+  private val primitiveFloatCase: Schema.Case[DynamicValue.Primitive[Float], DynamicValue] =
+    Schema.Case(
+      "Float",
+      Schema.primitive[Float].transform(f => DynamicValue.Primitive(f, StandardType[Float]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Float]]
+    )
+
+  private val primitiveDoubleCase: Schema.Case[DynamicValue.Primitive[Double], DynamicValue] =
+    Schema.Case(
+      "Double",
+      Schema.primitive[Double].transform(d => DynamicValue.Primitive(d, StandardType[Double]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Double]]
+    )
+
+  private val primitiveBinaryCase: Schema.Case[DynamicValue.Primitive[Chunk[Byte]], DynamicValue] =
+    Schema.Case(
+      "Binary",
+      Schema.primitive[Chunk[Byte]].transform(ch => DynamicValue.Primitive(ch, StandardType[Chunk[Byte]]), _.value),
+      _.asInstanceOf[DynamicValue.Primitive[Chunk[Byte]]]
```

Comment:
These tests won't work because of type erasure. I think we need to do it like in here: https://github.com/thinkharderdev/zio-schema/commit/7077f7bd5b6c426b08cc76a24c986ee8be3a6e16

Reply:
Yes, I got the same problem during testing
Will redo the casting according to your branch!